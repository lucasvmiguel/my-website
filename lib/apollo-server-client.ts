import { ApolloClient, InMemoryCache } from '@apollo/client'

export const client = new ApolloClient({
  uri: process.env.HASURA_APOLLO_CLIENT_URI,
  cache: new InMemoryCache(),
  headers: {
    'X-Hasura-Admin-Secret': process.env.HASURA_APOLLO_CLIENT_TOKEN as string
  }
})
