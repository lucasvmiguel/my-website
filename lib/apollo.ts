import { ApolloClient, InMemoryCache } from '@apollo/client'

export const client = new ApolloClient({
  cache: new InMemoryCache(),
  uri: `${process.env.NEXT_PUBLIC_BASE_URL}/api/graphql`
})
