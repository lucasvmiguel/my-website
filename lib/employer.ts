export interface Employer {
  name: string;
  description: string;
  image: string;
  jobTitle: string;
  tags: string[];
}

export const employers: Array<Employer> = [
  {
    name: 'Workday',
    description: 'Finance, HR, and planning',
    image: 'workday-logo.png',
    jobTitle: 'Software Engineer',
    tags: ['Kubernetes', 'Go', 'Helm', 'Istio', 'Terraform', 'Ruby', 'Ruby on Rails', 'Sidekiq', 'React', 'AWS', 'Docker']
  },
  {
    name: 'Noa',
    description: 'Journalism over audio',
    image: 'noa-logo.png',
    jobTitle: 'Software Engineer',
    tags: ['React', 'Redux', 'NodeJS', 'React Router', 'Google Analytics', 'Elasticsearch', 'MySQL', 'Express', 'Sequelize', 'Redux Form', 'AWS']
  },
  {
    name: 'B2W',
    description: 'E-commerce',
    image: 'b2w-logo.png',
    jobTitle: 'Software Engineer',
    tags: ['Go', 'Elasticsearch', 'Clojure', 'React', 'Redux', 'AngularJS', 'Microservices', 'Mongo', 'AWS', 'Docker']
  }
]
