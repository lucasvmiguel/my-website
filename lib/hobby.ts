import { IconType } from 'react-icons'
import { BiRun, BiCodeBlock } from 'react-icons/bi'
import { FaLuggageCart } from 'react-icons/fa'

export interface Hobby {
  title: string;
  description: string;
  icon: IconType;
}

export const hobbies: Array<Hobby> = [
  {
    title: 'Running',
    description: 'While I\'m listening music or a podcast is such a wonderful experience.',
    icon: BiRun
  },
  {
    title: 'Travel',
    description: 'Exploring the world and all its culture is something that always up to. My next destines are Egypt, Italy and Spain.',
    icon: FaLuggageCart
  },
  {
    title: 'Coding',
    description: 'Well, as you can see, I didn\'t need to have this website at all. However, I love coding in my free time.',
    icon: BiCodeBlock
  }
]
