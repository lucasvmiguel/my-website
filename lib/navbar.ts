export interface NavItem {
  label: string;
  subLabel?: string;
  children?: Array<NavItem>;
  href?: string;
}

export const navItems: Array<NavItem> = [
  {
    label: 'About me',
    href: '#'
  },
  {
    label: 'Articles',
    children: [
      {
        label: 'Article title 1',
        subLabel: 'Article description 1',
        href: '#'
      },
      {
        label: 'Article title 2',
        subLabel: 'Article description 2',
        href: '#'
      }
    ]
  }
]
