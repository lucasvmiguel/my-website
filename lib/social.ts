import { IconType } from 'react-icons'
import { FaLinkedin, FaTwitter } from 'react-icons/fa'
import { AiFillGithub } from 'react-icons/ai'

export interface SocialItem {
  label: string;
  icon: IconType;
  href: string;
}

export const socialItems: Array<SocialItem> = [
  {
    label: 'Github',
    icon: AiFillGithub,
    href: 'https://github.com/lucasvmiguel/'
  },
  {
    label: 'Linkedin',
    icon: FaLinkedin,
    href: 'https://www.linkedin.com/in/lucasvieiramiguel/'
  },
  {
    label: 'Twitter',
    icon: FaTwitter,
    href: 'https://twitter.com/LucasVieiraDev'
  }
]
