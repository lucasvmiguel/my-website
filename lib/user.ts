import { createUserMutation, findUserByExternalIdQuery } from '../graphql/user'

export const createOrFindUserIdByExternalId = async (externalId: string) => {
  const user = await findUserByExternalId(externalId)

  if (user) {
    return user.id
  }

  const { id } = await createUser(externalId)
  return id
}

const findUserByExternalId = async (externalId: string) => {
  const { data } = await findUserByExternalIdQuery(externalId)
  if (data.users.length > 0) {
    return data.users[0]
  }
  return null
}

const createUser = async (externalId: string) => {
  const { data: { insert_users_one: { id } } } = await createUserMutation('user', externalId)

  return { id }
}
