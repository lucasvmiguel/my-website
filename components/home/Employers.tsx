import { ReactNode } from 'react'
import {
  Box,
  Flex,
  Heading,
  Text,
  Stack,
  Container,
  Avatar,
  useColorModeValue,
  Tag
} from '@chakra-ui/react'

import { Employer } from '../../lib/employer'

const TestimonialContent = ({ children }: { children: ReactNode }) => {
  return (
    <Stack
      bg={useColorModeValue('white', 'gray.800')}
      boxShadow={'lg'}
      p={8}
      rounded={'xl'}
      align={'center'}
      pos={'relative'}
      _after={{
        content: '""',
        w: 0,
        h: 0,
        borderLeft: 'solid transparent',
        borderLeftWidth: 16,
        borderRight: 'solid transparent',
        borderRightWidth: 16,
        borderTop: 'solid',
        borderTopWidth: 16,
        borderTopColor: useColorModeValue('white', 'gray.800'),
        pos: 'absolute',
        bottom: '-16px',
        left: '50%',
        transform: 'translateX(-50%)'
      }}>
      {children}
    </Stack>
  )
}

const TestimonialHeading = ({ children }: { children: ReactNode }) => {
  return (
    <Heading as={'h3'} fontSize={'xl'}>
      {children}
    </Heading>
  )
}

const TestimonialText = ({ children }: { children: ReactNode }) => {
  return (
    <Text
      textAlign={'center'}
      color={useColorModeValue('gray.600', 'gray.400')}
      fontSize={'sm'}>
      {children}
    </Text>
  )
}

const TestimonialAvatar = ({
  src,
  name,
  title
}: {
  src: string;
  name: string;
  title: string;
}) => {
  return (
    <Flex align={'center'} mt={8} direction={'column'}>
      <Avatar src={src} alt={name} mb={2} />
      <Stack spacing={-1} align={'center'}>
        <Text fontWeight={600}>{name}</Text>
        <Text fontSize={'sm'} color={useColorModeValue('gray.600', 'gray.400')}>
          {title}
        </Text>
      </Stack>
    </Flex>
  )
}

interface EmployersProps {
  items: Employer[]
}

const Employers = ({ items }: EmployersProps) => {
  return (
    <Box bg={useColorModeValue('gray.100', 'gray.700')}>
      <Container maxW={'7xl'} py={16} as={Stack} spacing={12}>
        <Stack spacing={0} align={'center'}>
          <Heading>Employers</Heading>
          <Text>Companies that trust in me</Text>
        </Stack>
        <Stack
          direction={{ base: 'column', md: 'row' }}
          spacing={{ base: 10, md: 4, lg: 10 }}>
          {items.map((item, i) => (
            <Box key={i}>
              <TestimonialContent>
                <TestimonialHeading>{item.jobTitle}</TestimonialHeading>
                <TestimonialText>
                  {item.tags.map((tag, i) => <Tag m={1} key={i} bg={'green.400'}>{tag}</Tag>)}
                </TestimonialText>
              </TestimonialContent>
              <TestimonialAvatar
                src={
                  item.image
                }
                name={item.name}
                title={item.description}
              />
            </Box>
          ))}
        </Stack>
      </Container>
    </Box>
  )
}

export default Employers
