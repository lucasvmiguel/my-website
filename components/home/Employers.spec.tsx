import renderer from 'react-test-renderer'

import Employers from './Employers'

import { employers } from '../../lib/employer'

it('renders Employers component correctly - snapshot', () => {
  const tree = renderer
    .create(<Employers items={employers} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
