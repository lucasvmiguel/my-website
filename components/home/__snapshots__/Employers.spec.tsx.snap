// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Employers component correctly - snapshot 1`] = `
<div
  className="css-1mkcx3b"
>
  <div
    className="chakra-stack chakra-container css-lxy0v2"
  >
    <div
      className="chakra-stack css-jp20jv"
    >
      <h2
        className="chakra-heading css-0"
      >
        Employers
      </h2>
      <p
        className="chakra-text css-0"
      >
        Companies that trust in me
      </p>
    </div>
    <div
      className="chakra-stack css-1itflco"
    >
      <div
        className="css-0"
      >
        <div
          className="chakra-stack css-e5fz58"
        >
          <h3
            className="chakra-heading css-6dvxm2"
          >
            Software Engineer
          </h3>
          <p
            className="chakra-text css-1cuin5z"
          >
            <span
              className="css-1sndsea"
            >
              Kubernetes
            </span>
            <span
              className="css-1sndsea"
            >
              Go
            </span>
            <span
              className="css-1sndsea"
            >
              Helm
            </span>
            <span
              className="css-1sndsea"
            >
              Istio
            </span>
            <span
              className="css-1sndsea"
            >
              Terraform
            </span>
            <span
              className="css-1sndsea"
            >
              Ruby
            </span>
            <span
              className="css-1sndsea"
            >
              Ruby on Rails
            </span>
            <span
              className="css-1sndsea"
            >
              Sidekiq
            </span>
            <span
              className="css-1sndsea"
            >
              React
            </span>
            <span
              className="css-1sndsea"
            >
              AWS
            </span>
            <span
              className="css-1sndsea"
            >
              Docker
            </span>
          </p>
        </div>
        <div
          className="css-w6zqsg"
        >
          <span
            alt="Workday"
            className="chakra-avatar css-ggnqxp"
          >
            <svg
              aria-label=" avatar"
              className="chakra-avatar__svg css-16ite8i"
              role="img"
              viewBox="0 0 128 128"
            >
              <path
                d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                fill="currentColor"
              />
              <path
                d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            className="chakra-stack css-1rreaqf"
          >
            <p
              className="chakra-text css-35ezg3"
            >
              Workday
            </p>
            <p
              className="chakra-text css-130t7v7"
            >
              Finance, HR, and planning
            </p>
          </div>
        </div>
      </div>
      <div
        className="css-0"
      >
        <div
          className="chakra-stack css-e5fz58"
        >
          <h3
            className="chakra-heading css-6dvxm2"
          >
            Software Engineer
          </h3>
          <p
            className="chakra-text css-1cuin5z"
          >
            <span
              className="css-1sndsea"
            >
              React
            </span>
            <span
              className="css-1sndsea"
            >
              Redux
            </span>
            <span
              className="css-1sndsea"
            >
              NodeJS
            </span>
            <span
              className="css-1sndsea"
            >
              React Router
            </span>
            <span
              className="css-1sndsea"
            >
              Google Analytics
            </span>
            <span
              className="css-1sndsea"
            >
              Elasticsearch
            </span>
            <span
              className="css-1sndsea"
            >
              MySQL
            </span>
            <span
              className="css-1sndsea"
            >
              Express
            </span>
            <span
              className="css-1sndsea"
            >
              Sequelize
            </span>
            <span
              className="css-1sndsea"
            >
              Redux Form
            </span>
            <span
              className="css-1sndsea"
            >
              AWS
            </span>
          </p>
        </div>
        <div
          className="css-w6zqsg"
        >
          <span
            alt="Noa"
            className="chakra-avatar css-ggnqxp"
          >
            <svg
              aria-label=" avatar"
              className="chakra-avatar__svg css-16ite8i"
              role="img"
              viewBox="0 0 128 128"
            >
              <path
                d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                fill="currentColor"
              />
              <path
                d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            className="chakra-stack css-1rreaqf"
          >
            <p
              className="chakra-text css-35ezg3"
            >
              Noa
            </p>
            <p
              className="chakra-text css-130t7v7"
            >
              Journalism over audio
            </p>
          </div>
        </div>
      </div>
      <div
        className="css-0"
      >
        <div
          className="chakra-stack css-e5fz58"
        >
          <h3
            className="chakra-heading css-6dvxm2"
          >
            Software Engineer
          </h3>
          <p
            className="chakra-text css-1cuin5z"
          >
            <span
              className="css-1sndsea"
            >
              Go
            </span>
            <span
              className="css-1sndsea"
            >
              Elasticsearch
            </span>
            <span
              className="css-1sndsea"
            >
              Clojure
            </span>
            <span
              className="css-1sndsea"
            >
              React
            </span>
            <span
              className="css-1sndsea"
            >
              Redux
            </span>
            <span
              className="css-1sndsea"
            >
              AngularJS
            </span>
            <span
              className="css-1sndsea"
            >
              Microservices
            </span>
            <span
              className="css-1sndsea"
            >
              Mongo
            </span>
            <span
              className="css-1sndsea"
            >
              AWS
            </span>
            <span
              className="css-1sndsea"
            >
              Docker
            </span>
          </p>
        </div>
        <div
          className="css-w6zqsg"
        >
          <span
            alt="B2W"
            className="chakra-avatar css-ggnqxp"
          >
            <svg
              aria-label=" avatar"
              className="chakra-avatar__svg css-16ite8i"
              role="img"
              viewBox="0 0 128 128"
            >
              <path
                d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                fill="currentColor"
              />
              <path
                d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                fill="currentColor"
              />
            </svg>
          </span>
          <div
            className="chakra-stack css-1rreaqf"
          >
            <p
              className="chakra-text css-35ezg3"
            >
              B2W
            </p>
            <p
              className="chakra-text css-130t7v7"
            >
              E-commerce
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
